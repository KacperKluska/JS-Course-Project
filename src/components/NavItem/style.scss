.navItem {
  color: white;
  cursor: pointer;
}

@media (max-width: 550px) {
  .navItem {
    padding: 0.5rem;
  }
}
