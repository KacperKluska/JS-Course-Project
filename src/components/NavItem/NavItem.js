import './style.scss';

export default function NavItem({ icon, children, onClick }) {
  return (
    <li className="navItem">
      <button type="button" onClick={onClick}>
        <i className={icon} />
        {children}
      </button>
    </li>
  );
}
