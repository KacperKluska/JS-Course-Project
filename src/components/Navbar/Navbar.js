import { Link } from "react-router-dom";
import NavItem from "../NavItem/NavItem";
import CustomLink from "../CustomLink/CustomLink";
import "./style.scss";

export default function Navbar() {
  const scrollToElById = (id) => {
    setTimeout(() => {
      const element = document.getElementById(`${id}`);
      if (element === null) {
        return;
      }
      const offset = element.offsetTop;
      window.scrollTo({ top: offset - 50, behavior: "smooth" });
    }, 0);
  };

  /**
   * TOASK
   * How can I do it with useRef ?
   */

  return (
    <nav className="sticky">
      <div className="secondNavbar">
        <ul className="thirdList">
          <NavItem>For him</NavItem>
          <NavItem>For her</NavItem>
          <NavItem>Kids</NavItem>
          <NavItem onClick={() => scrollToElById("About")}>
            <Link className="navbarLink" to="/">
              About
            </Link>
          </NavItem>
          <NavItem onClick={() => scrollToElById("Contact")}>
            <Link className="navbarLink" to="/">
              Contact
            </Link>
          </NavItem>
        </ul>
      </div>
    </nav>
  );
}
