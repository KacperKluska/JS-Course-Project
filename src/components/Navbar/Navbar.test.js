import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Navbar from "./Navbar";
import NavItem from "../NavItem/NavItem";

let wrapper;

beforeAll(() => {
  wrapper = shallow(<Navbar />);
});

test("checking blocks", () => {
  expect(wrapper.find(".sticky").html()).toContain('<div id="secondNavbar">');
  expect(wrapper.find("#secondNavbar").html()).toContain('<ul id="thirdList">');
});

test("checking list of navitems", () => {
  expect(wrapper.find("NavItem").length).toEqual(5);
  //   wrapper.find("NavItem").forEach((item) => {
  //     expect(item.type()).toBe("NavItem");
  //   });
  /**
   * TODO
   * try to do this with render or mount
   */
  expect(wrapper.find("NavItem").get(0)).toStrictEqual(
    <NavItem>For him</NavItem>
  );
  expect(wrapper.find("NavItem").get(1)).toStrictEqual(
    <NavItem>For her</NavItem>
  );
  expect(wrapper.find("NavItem").get(2)).toStrictEqual(<NavItem>Kids</NavItem>);
  expect(wrapper.find("NavItem").get(3)).toStrictEqual(
    <NavItem>About</NavItem>
  );
  expect(wrapper.find("NavItem").get(4)).toStrictEqual(
    <NavItem>Contact</NavItem>
  );
});

test("Navbar snapshot", () => {
  const container = renderer.create(<Navbar />);
  expect(container.toJSON()).toMatchSnapshot();
});
