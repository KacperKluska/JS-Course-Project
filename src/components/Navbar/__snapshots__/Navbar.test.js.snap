// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar snapshot 1`] = `
<nav
  className="sticky"
>
  <div
    className="secondNavbar"
  >
    <ul
      className="thirdList"
    >
      <li
        className="navItem"
      >
        <button
          className="navItemButton"
          type="button"
        >
          <i />
          For him
        </button>
      </li>
      <li
        className="navItem"
      >
        <button
          className="navItemButton"
          type="button"
        >
          <i />
          For her
        </button>
      </li>
      <li
        className="navItem"
      >
        <button
          className="navItemButton"
          type="button"
        >
          <i />
          Kids
        </button>
      </li>
      <li
        className="navItem"
      >
        <button
          className="navItemButton"
          onClick={[Function]}
          type="button"
        >
          <i />
          <a
            className="navbarLink"
            href="/"
            onClick={[Function]}
          >
            About
          </a>
        </button>
      </li>
      <li
        className="navItem"
      >
        <button
          className="navItemButton"
          onClick={[Function]}
          type="button"
        >
          <i />
          <a
            className="navbarLink"
            href="/"
            onClick={[Function]}
          >
            Contact
          </a>
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
